### PR TITLE
Separate create and update routes, minor housekeeping

### PR DESCRIPTION
**Pull request by tkyi**
Separated out the create and update calls into POST and PUT, respectively

- The create call generates an `id` based on the `name` and `version` of the platform (using screwdriver-hashr) and then saves it to the datastore
- The update now calls `datastore.update` for the specified platform
- Also lowercased all module names

Using: https://github.com/screwdriver-cd/hashr/pull/1
Very similar to changes made to builds plugin: https://github.com/screwdriver-cd/plugin-builds/pull/12

### CODE REVIEW

File: lib/create.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const urlLib = require('url');
+
+const dataModel = require('screwdriver-data-model');
+const createModel = dataModel.platform.create;
+
+module.exports = (datastore) => ({
+    method: 'POST',
+    path: '/platforms',
+    config: {
+        description: 'Save a platform',
```
Comment (d2lam) [inline lib/create.js]: Change this to 'Create'

**Comment by d2lam** [comment]
small comment. The rest lgtm 👍

**Comment by d2lam** [comment]
👍 
200w-4

File: lib/update.js [hunk 2]
```diff
@@ -14,31 +15,33 @@ module.exports = (datastore) => ({
...
 
             // eslint-disable-next-line consistent-return
-            datastore.get(config, (err, data) => {
+            datastore.update(config, (err, data) => {
                 if (err) {
                     return reply(err);
                 }
                 if (!data) {
-                    return reply(Boom.notFound('Platform does not exist'));
+                    return reply(boom.notFound('Platform ${id} does not exist'));
```
Comment (nkatzman) [inline lib/update.js]: This string interpolation requires backticks instead of quotes

File: package.json [hunk 3]
```diff
@@ -37,12 +37,15 @@
...
     "sinon": "^1.17.4"
   },
   "dependencies": {
     "boom": "^3.2.2",
     "hoek": "^4.0.1",
     "joi": "^8.4.2",
-    "screwdriver-data-model": "^1.0.5"
+    "screwdriver-data-model": "^1.0.5",
+    "screwdriver-hashr": "^1.0.0",
+    "url": "^0.11.0"
```
Comment (FenrirUnbound) [inline package.json]: ?

Comment (FenrirUnbound) [inline package.json]: `url` is part of the core NodeJS language: [Source](https://nodejs.org/dist/latest-v4.x/docs/api/url.html)

File: package.json [hunk 4]
```diff
@@ -37,12 +37,15 @@
...
     "jenkins-mocha": "^2.6.0",
+    "mockery": "^1.7.0",
     "sinon": "^1.17.4"
   },
   "dependencies": {
     "boom": "^3.2.2",
     "hoek": "^4.0.1",
     "joi": "^8.4.2",
-    "screwdriver-data-model": "^1.0.5"
+    "screwdriver-data-model": "^1.0.5",
```
Comment (d2lam) [inline package.json]: We should update this to ^2.0.0
Just a warning it's going to fail all the old tests.

Comment (d2lam) [inline package.json]: Or in a different PR. idc. up to you!

**Comment by nkatzman** [comment]
👍

**Pull request merged**